contracts/gauge-borrowing/VaultPriceOracle.sol

```diff
@@ -0,0 +1,185 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/utils/math/SafeMathUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+
+import "../interfaces/Compound/IPriceOracle.sol";
+import "../interfaces/Compound/IBasePriceOracle.sol";
+import "../interfaces/Compound/ICToken.sol";
+import "../interfaces/Compound/ICErc20.sol";
+import "../interfaces/Compound/IAggregatorV3Interface.sol";
+import "../interfaces/Compound/ILiquidityGauge.sol";
+import "../interfaces/Compound/IRibbonVault.sol";
+
+/**
+ * @title VaultPriceOracle
+ * @notice Returns prices from Chainlink.
+ * @dev Implements `PriceOracle`.
+ * @author David Lucid <david@rari.capital> (https://github.com/davidlucid)
+ */
+contract VaultPriceOracle is IPriceOracle, IBasePriceOracle {
```

Comment:
basing off of david lucid's ChainlinkPriceOracleV2 which is used in the MasterPriceOracle

Reply:
The original contract `ChainlinkPriceOracleV2` was audited by QuantStamp / Rari

https://certificate.quantstamp.com/full/rari-capital-on-chain-governance

```
QSP-1 ChainlinkPriceOracleV2._price Does Not Consult Decimal Values
Severity: Medium Risk
Status: Fixed
File(s) affected: ChainlinkPriceOracleV2.sol
The function queries various Chainlink oracles, however, the decimal value associated with each price feed is not considered. This may lead to over/undervaluing the price
of the token being considered.
```

```
ChainlinkPriceOracleV2.changeAdmin should ensure that newAdmin is non-zero, unless this is intended to revoke privileged roles.
```